WHY_IBC.md

```diff
@@ -0,0 +1,49 @@
+## Why IBC?
+
+The design space of "interblockchain communication protocols" is wide, and the term itself has become a bit too all-encompassing. The "Interblockchain Communication Protocol" (IBC) is a very particular point in that design space, chosen to provide specific security, composability, and modularity properties for the expected interchain ecosystem of interoperable blockchains. This document outlines the "why" of IBC and enumerates the primary high-level design goals.
+
+### Heterogeneity
+
+- Supports heterogeneous chains with different state machines
+- Once across the IBC interface barrier, state machines can do whatever they want (e.g. shield tokens)
+
+### Locality
+
+- Assumptions are designed to be informationally "local"
+- Chains must only understand state of chains to which they are connected
+- No necessary single root chain, dynamic network topology, no protocol-level scaling limitations
+- Reflects local nature of underlying global commerce (frequency of transactions falls off over distance)
+
+#### Locality of communication & information
+
+- No global topology view required
+- Core protocol can be reasoned about as a construction between two chains, routing built on top
+- Users and chains can reason about security guarantees given what they know and trust
+
+#### Locality of trust & security
+
+- Users of IBC choose which consensus algorithms & validator sets they trust
```

Comment:
Nit: The term users might be a bit vague here. Are those end-users, ledgers/chains?